x/bank/legacy/v0_40/migrate.go

```diff
@@ -2,21 +2,36 @@ package v040
 
 import (
 	v039auth "github.com/cosmos/cosmos-sdk/x/auth/legacy/v0_39"
+	v036supply "github.com/cosmos/cosmos-sdk/x/bank/legacy/v0_36"
 	v038bank "github.com/cosmos/cosmos-sdk/x/bank/legacy/v0_38"
+	v040bank "github.com/cosmos/cosmos-sdk/x/bank/types"
```

Comment:
note to @alexanderbez: I remember you wanting the migrate scripts to be independent of the actual module's `types/` folder.

I initially went down this road, but it created some problems:
1. If I copied the `bank.pb.go` and `genesis.pb.go` files into `x/banks/legacy/v0_40`, then protobuf would show error "duplicate proto registration" because it registers each type twice, which pollutes the migrate command stdout.
2. If I copied some parts of `bank.pb.go` and `genesis.pb.go` into `x/banks/legacy/v0_40` (e.g. not copying the duplicate proto registration, or only copying the structs and some methods), that would involve a lot of work, and tweaking generated proto files, which is very prone to human errors.

So at Regen yesterday, we decided to just use the `types` from the modules:
- with these proto files, we're aiming at having very few breaking changes, which means that future migrate scripts can use the same proto definitions.
- if there's really a breaking change, then we'll copy the `.pb.go` into `legacy/` folders, but there won't be "duplicate proto registration" as we're registering under a new proto package name.